django/db/models/lookups.py

```diff
@@ -463,6 +465,12 @@ class IsNull(BuiltinLookup):
     prepare_rhs = False
 
     def as_sql(self, compiler, connection):
+        if not isinstance(self.rhs, bool):
+            warnings.warn(
+                'Providing a non-bool value with a isnull lookup is untested and '
+                'undocumented leading to inconsistent behaviour.',
```

Comment:
Maybe:
```
'Using a non-boolean value with an isnull lookup is deprecated, '
'use True or False.'
```
I don't think that we need to modify `stacklevel`. We should also add a comment that will make easier to remove this in Django 4.0, e.g.
```python
if not isinstance(self.rhs, bool):
    # When the deprecation ends, replace with:
    # raise ValueError(
    #     'The QuerySet value for an isnull lookup must be True or '
    #     'False.'
    # )
    warnings.warn(
        'Using a non-boolean value with an isnull lookup is '
        'deprecated, use True or False instead.',
        RemovedInDjango40Warning,
    )
```